app/controllers/preferences_controller.rb

```diff
@@ -1,8 +1,15 @@
-# frozen_string_literal: true
-
 class PreferencesController < ApplicationController
   def index
     @preferences = current_user.preferences
     @pagy, @records = pagy(@preferences)
   end
+
+  def new
+    @preference = Preference.new
+  end
+
+  def create
+    preference_params = params.require(:preference).permit(:name, :description, :restriction)
+    @preference = current_user.preferences.build(preference_params)
```

Comment:
Would be nice to have a flash message to show the user the success or not of the action, wdyt?